src/formatDate.js

```diff
@@ -45,12 +45,67 @@
  * @param {string} date
  * @param {string[]} fromFormat
  * @param {string[]} toFormat
- *
+ *formatDate.js
  * @returns {string}
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let newFormatDate = '';
+  let newDay = '';
+  let newMonth = '';
+  let newYear = '';
+  const newSeparator = toFormat[3];
+
+  let oldDateSplit = [];
+  let oldYear = '';
+
+  oldDateSplit = date.split(fromFormat[3]);
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'DD') {
+      newDay = oldDateSplit[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      newMonth = oldDateSplit[i];
+    }
+
+    if (fromFormat[i].slice(0, 2) === 'YY') {
+      oldYear = oldDateSplit[i];
+    }
+  }
+
+  for (const elem of toFormat) {
+    if (elem === 'DD') {
+      newFormatDate += (newDay + newSeparator);
+    }
+
+    if (elem === 'MM') {
+      newFormatDate += (newMonth + newSeparator);
+    }
+
+    if (elem.slice(0, 2) === 'YY') {
+      if (elem.length === oldYear.length) {
+        newYear = oldYear;
+      }
+
+      if (elem.length < oldYear.length) {
+        newYear = oldYear.slice(2, 4);
+      }
+
+      if (elem.length > oldYear.length) {
+        if (oldYear < '30') {
```

Comment:
The comparison of 'oldYear' with '30' is incorrect. It should be compared as a number, not as a string. Convert 'oldYear' to a number before the comparison.